### PR TITLE
Added presubmit script distilled from that in trillian

### PR DESCRIPTION
**Pull request by mhutchinson**
Removed some of the flags and branches that weren't needed, such as code
generation. This isn't hooked into CI _yet_, but at least developers can
run `./scripts/presubmit.sh` before submitting in the meantime.

### CODE REVIEW

File: scripts/check_license.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```
Comment (pavelkalinnikov) [inline scripts/check_license.sh:1]: Should this file itself have a license header, btw? :D

**Comment by pavelkalinnikov** [comment]
Do you recon this script is github-actionable?

File: scripts/check_license.sh (hunk 1 above)
Comment (mhutchinson) [reply scripts/check_license.sh:1]: There isn't one in the other shell scripts, nor other supporting files (e.g. Dockerfile). As a) I'm probably less of an expert than Rob that wrote this stuff, and b) consistency, I'm strongly inclined not to start adding stuff here.

Comment (pavelkalinnikov) [reply scripts/check_license.sh:1]: I'm inclined to agree.

**Comment by mhutchinson** [comment]
> Do you reckon this script is github-actionable?

I'm planning on making it consistent with the other repos, which have GCB run a docker container that performs all of the checks. If we foresee only ever wanting then maybe we could look at https://github.com/golangci/golangci-lint-action.

**Comment by pavelkalinnikov** [comment]
Consistency across is the best thing, SGTM.

I like the simplicity of actions, it worked well for this `merkle` repo for adding quick easy checks. I don't know if it can fully replace GCB though. Maybe the latter can be used for more involved integration tests.

**Review by pavelkalinnikov — APPROVED** [review]

**Pull request merged**